app/main/db/queries.py

```diff
@@ -125,12 +129,43 @@ def build_fuzzy_search_summary_query(query_string: str):
             Consignment.SeriesId == Series.SeriesId,
         )
         .join(File, File.ConsignmentId == Consignment.ConsignmentId)
-        .join(FileMetadata, File.FileId == FileMetadata.FileId, isouter=True)
-        .where(and_(func.lower(File.FileType) == "file", fuzzy_filters))
         .group_by(Body.BodyId)
         .order_by(Body.Name)
+    ).subquery()
+
+    query = (
+        db.session.query(
+            sub_query.c.transferring_body_id,
+            sub_query.c.transferring_body,
+            func.count(sub_query.c.records_held).label("records_held"),
+        )
+        .join(Series, Series.BodyId == sub_query.c.transferring_body_id)
+        .join(
+            Consignment,
+            Consignment.SeriesId == Series.SeriesId,
+        )
+        .join(File, File.ConsignmentId == Consignment.ConsignmentId)
+        .where(func.lower(File.FileType) == "file")
+        .group_by(
+            sub_query.c.transferring_body_id, sub_query.c.transferring_body
+        )
     )
 
+    search_terms = _get_search_terms(query_string)
+    if len(search_terms) > 0:
+        for term in search_terms:
```

Comment:
in both build_fuzzy_search_summary_query and build_fuzzy_search_transferring_body_query we do this same filtering for loop/ Can we reduce this duplication so that any further changes to this only needs to be done in 1 place / less likely to introduce bugs etc.